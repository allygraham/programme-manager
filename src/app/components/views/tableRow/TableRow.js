import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

const Row = styled('tr')`
  background-color: ${props => props.active ? '#FFF' : '#D5D5D5'};
  cursor: pointer;
  z-index: 1;
`;

const RemoveButton = styled('td')`
  cursor: pointer;
  transition: color 300ms ease-in-out;
  z-index: 5;

  &:hover {
    color: #B00020;
  }
`;

export const TableRow = ({ programme, handleProgrammeRemove, handleProgrammeEdit }) => (
  <Row
    active={programme.active}
  >
    <td onClick={() => handleProgrammeEdit(programme.id)}>{programme.id}</td>
    <td onClick={() => handleProgrammeEdit(programme.id)}>{programme.name}</td>
    <td onClick={() => handleProgrammeEdit(programme.id)}>{programme.shortDescription}</td>
    <td onClick={() => handleProgrammeEdit(programme.id)}>{programme.active ? 'Active' : 'Inactive'}</td>
    <RemoveButton
      className="has-text-centered"
      onClick={() => handleProgrammeRemove(programme.id)}
    >
      <i className="fa fa-close"/>
    </RemoveButton>
  </Row>
);

TableRow.propTypes = {
  programme: PropTypes.object.isRequired,
  handleProgrammeRemove: PropTypes.func.isRequired,
  handleProgrammeEdit: PropTypes.func.isRequired,
};