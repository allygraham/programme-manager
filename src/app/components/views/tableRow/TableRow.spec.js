import React from 'react';
import { shallow } from 'enzyme';

import { TableRow } from './TableRow';

describe('TableRow', () => {

  test('render the <TableRow> component', () => {
    const handleProgrammeRemove = jest.fn();
    const handleProgrammeEdit = jest.fn();
    const programme = {};

    const wrapper = shallow(<TableRow handleProgrammeRemove={handleProgrammeRemove} handleProgrammeEdit={handleProgrammeEdit} programme={programme} />);
    expect(wrapper).toMatchSnapshot();
  });

});