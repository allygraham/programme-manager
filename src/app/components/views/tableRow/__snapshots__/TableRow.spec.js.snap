// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow render the <TableRow> component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableRow
    handleProgrammeEdit={[MockFunction]}
    handleProgrammeRemove={[MockFunction]}
    programme={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "active": undefined,
      "children": Array [
        <td
          onClick={[Function]}
        />,
        <td
          onClick={[Function]}
        />,
        <td
          onClick={[Function]}
        />,
        <td
          onClick={[Function]}
        >
          Inactive
        </td>,
        <Styled(td)
          className="has-text-centered"
          onClick={[Function]}
        >
          <i
            className="fa fa-close"
          />
        </Styled(td)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Inactive",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Inactive",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <i
            className="fa fa-close"
          />,
          "className": "has-text-centered",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-close",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": undefined,
        "children": Array [
          <td
            onClick={[Function]}
          />,
          <td
            onClick={[Function]}
          />,
          <td
            onClick={[Function]}
          />,
          <td
            onClick={[Function]}
          >
            Inactive
          </td>,
          <Styled(td)
            className="has-text-centered"
            onClick={[Function]}
          >
            <i
              className="fa fa-close"
            />
          </Styled(td)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Inactive",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Inactive",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <i
              className="fa fa-close"
            />,
            "className": "has-text-centered",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-close",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
